jupyterhub/values.yaml

```diff
@@ -142,6 +142,17 @@ auth:
 
 
 singleuser:
+  extraTolerations: []
+  nodeSelector: {}
+  extraNodeAffinity:
+    required: []
+    preferred: []
+  extraPodAffinity:
+    required: []
+    preferred: []
+  extraPodAntiAffinity:
+    required: []
+    preferred: []
```

Comment:
Can we make this expose the Kubernetes `affinity` structure directly, instead of exposing six different configurables? It seems that would be a lot simpler, both for us to maintain and for users to understand. In general, let's try *very hard* to avoid using anything but Kubernetes' own names and structures for things.

Reply:
So I have considered this *in depth* now, and the conclusion is no, not with Helm 2, in the values section. If we do, we wont be able to augment our configurable presets within the templates.

But, in kubespawner, that is configured through python where we have great capabilities of working with the stuff coming from the configmap, we could combine them again and set one big affinity traitlet on kubespawner.

I was confused about this until recently. In summary, this PR must be like this, but the previous one for kubespawner did not have to be that way =/